From 093d237a60a1960e3d74dd0ca34c7a0f6716ed0e Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Thu, 24 Mar 2022 15:17:12 -0300
Subject: [PATCH 028/163] stdlib: Fix tst-getrandom memcmp call

The idea is to check if the up sizeof (buf) are equal, not only
the first byte.

Checked on x86_64-linux-gnu and i686-linux-gnu.

(cherry picked from commit 7f2ddf7400bb959897a5fe58f7fc5fbe5e57cfae)
---
 stdlib/tst-getrandom.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/stdlib/tst-getrandom.c b/stdlib/tst-getrandom.c
index 608fc4746e..61c3cae92e 100644
--- a/stdlib/tst-getrandom.c
+++ b/stdlib/tst-getrandom.c
@@ -189,7 +189,7 @@ test_getentropy (void)
 
   /* The probability that these two buffers are equal is very
      small. */
-  if (memcmp (buf, buf2, sizeof (buf) == 0))
+  if (memcmp (buf, buf2, sizeof (buf)) == 0)
     {
       printf ("error: getentropy appears to return constant bytes\n");
       errors = true;
-- 
2.37.2

